src/registries/npm-registry.js

```diff
@@ -47,7 +46,7 @@ export default class NpmRegistry extends Registry {
   }
 
   request(pathname: string, opts?: RegistryRequestOptions = {}): Promise<?Object> {
-    const registry = removeSuffix(String(this.registries.yarn.getOption('registry')), '/');
```

Comment:
instead of removing this line, it might be better to keep it and change the `url.resolve` line as ```
```
`${registry}/${pathname}`
```